packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/create-model.ts

```diff
@@ -0,0 +1,258 @@
+    }
+    return this._role;
+  }
+
+  public get grantPrincipal(): iam.IPrincipal {
+    if (this._grantPrincipal === undefined) {
+      throw new Error('Principal not available yet--use the object in a Task first');
+    }
+    return this._grantPrincipal;
+  }
+
+  /**
+   * Add the security group to all instances via the launch configuration
+   * security groups array.
+   *
+   * @param securityGroup: The security group to add
+   */
+  public addSecurityGroup(securityGroup: ec2.ISecurityGroup): void {
+    this.securityGroups.push(securityGroup);
+  }
+
+  /**
+   * @internal
+   */
+  protected _renderTask(): any {
+    return {
+      Resource: integrationResourceArn('sagemaker', 'createModel', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject(this.renderParameters()),
+    };
+  }
+
+  private renderParameters(): { [key: string]: any } {
+    return {
+      EnableNetworkIsolation: this.props.enableNetworkIsolation,
+      ExecutionRoleArn: this._role!.roleArn,
+      ModelName: this.props.modelName,
+      ...this.renderContainers(this.props.containers),
+      ...this.renderPrimaryContainer(this.props.primaryContainer),
+      ...this.renderTags(this.props.tags),
+      ...this.renderVpcConfig(this.props.vpcConfig),
+    };
+  }
+
+  private makePolicyStatements(): iam.PolicyStatement[] {
+    // set the SageMaker role or create new one
+    this._grantPrincipal = this._role =
+        this.props.role ||
+        new iam.Role(this, 'SagemakerRole', {
+          assumedBy: new iam.ServicePrincipal('sagemaker.amazonaws.com'),
+          inlinePolicies: {
+            CreateModel: new iam.PolicyDocument({
+              statements: [
+                new iam.PolicyStatement({
+                  actions: [
+                    'cloudwatch:PutMetricData',
+                    'logs:CreateLogStream',
+                    'logs:PutLogEvents',
+                    'logs:CreateLogGroup',
+                    'logs:DescribeLogStreams',
+                    's3:GetObject',
+                    'ecr:GetAuthorizationToken',
+                    'ecr:BatchCheckLayerAvailability',
+                    'ecr:GetDownloadUrlForLayer',
+                    'ecr:BatchGetImage',
+                    ...(this.props.vpcConfig
+                      ? [
+                        'ec2:CreateNetworkInterface',
+                        'ec2:CreateNetworkInterfacePermission',
+                        'ec2:DeleteNetworkInterface',
+                        'ec2:DeleteNetworkInterfacePermission',
+                        'ec2:DescribeNetworkInterfaces',
+                        'ec2:DescribeVpcs',
+                        'ec2:DescribeDhcpOptions',
+                        'ec2:DescribeSubnets',
+                        'ec2:DescribeSecurityGroups',
+                      ]
+                      : []),
+                  ],
+                  resources: ['*'], // Those permissions cannot be resource-scoped
```

Comment:
we should specify exactly why. these are a lot of APIs and I'm not sure why we would want `*` permissions to `putLogEvents`, `getObject`, etc

Reply:
I'll try to reduce permissions wherever possible based on https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html . 
while I see the way to scope `getObject` using modelDataUrl path if given and not json path. I'm not sure not to scope 
`putLogEvents `.  In the recommendation of how to narrow permissions for CreateModel API on the mentioned above link the following actions still have `*`
```
            "Effect": "Allow",
            "Action": [
                "cloudwatch:PutMetricData",
                "logs:CreateLogStream",
                "logs:PutLogEvents",
                "logs:CreateLogGroup",
                "logs:DescribeLogStreams",
                "ecr:GetAuthorizationToken"
            ],
            "Resource": "*"
```

Reply:
we can split it up to multiple statements if needed. I think the desired outcome is that the least amount of privileges required are added. We should limit operations to the resources that are referenced here. 

If the API itself does not support scoping down operations on a resource, then we would need to contain those in a statement.

Reply:
After I changing `modelDataUrl` type to `S3Location`, S3Location now is responsible for scoping the permissions. Same for the most ecr permissions, they are scoped in `DockerImage` type. For the permissions that are left, I regrouped them